test/jdk/java/lang/management/ThreadMXBean/Locks.java

```diff
@@ -362,15 +366,20 @@ public static void main(String args[]) throws Exception {
                 t2.start();
 
                 p.arriveAndAwaitAdvance(); // Phase 1 (blocking)
+
                 assertThreadState(t2, Thread.State.BLOCKED);
-                checkBlockedObject(t2, OBJC, mainThread);
+                if (!mainThread.isVirtual()) {
```

Comment:
Although the you covered the reason for this exclusion in the PR description, I think a comment is warranted here, and also for line 380 below.